File: Algorithms/Searching/Linear_Search/Linear_Search.py
# The below given code implements Linear Search algorithm.
# The program prints the index number of the element if the searched element is present in the list, else it prints -1.

def LinearSearch(arr, element):
    for i in range (len(arr)):
        if arr[i] == element:
            return i
    return -1

A = list(map(int, input("Enter the elements of list: ").split()))
searchKey = int(input("Enter element to be searched: "))
print(LinearSearch(A, searchKey))

File: Algorithms/Searching-Algorithm/Linear-Search/Linearsearch.c
/*
 * C program to input N numbers and store them in an array.
 * Do a linear search for a given key and report success
 * or failure.
 */
#include <stdio.h>
 
void main()
{  int num;
 
    int i,  keynum, found = 0;
 
    printf("Enter the number of elements ");
    scanf("%d", &num);
    int array[num];
    printf("Enter the elements one by one \n");
    for (i = 0; i < num; i++)
    {
        scanf("%d", &array[i]);
    }
 
    printf("Enter the element to be searched ");
    scanf("%d", &keynum);
    /*  Linear search begins */
    for (i = 0; i < num ; i++)
    {
        if (keynum == array[i] )
        {
            found = 1;
            break;
        }
    }
    if (found == 1)
        printf("Element is present in the array at position %d",i+1);
    else
        printf("Element is not present in the array\n");
}
